projects/aca-shared/rules/src/app.rules.ts

```diff
@@ -483,6 +483,48 @@ export function canOpenWithOffice(context: AcaRuleContext): boolean {
   return context.permissions.check(file, ['update']);
 }
 
+/**
+ * Checks if user savedSearches are supported by current ACS version.
+ * JSON ref: `isSavedSearchAvailable`
+ */
+export const isSavedSearchAvailable = createVersionRule('25.1.0');
+
+/**
+ * Partially applies minimal version of a feature against a core compatibility evaluation.
+ * @param minimalVersion The minimal version to check against.
+ */
+export function createVersionRule(minimalVersion: string): (context: RuleContext) => boolean {
+  return (context: RuleContext): boolean => {
+    const acsVersion = context.repository.version?.display?.split(' ')[0];
+    return isVersionCompatible(acsVersion, minimalVersion);
+  };
+}
+
+export function isVersionCompatible(currentVersion: string, minimalVersion: string): boolean {
+  if (!currentVersion || !minimalVersion) {
+    return false;
+  }
+
+  const currentParts = currentVersion.split('.').map(Number);
+  const minimalParts = minimalVersion.split('.').map(Number);
+  const maxLength = Math.max(currentParts.length, minimalParts.length);
+
+  for (let i = 0; i < maxLength; i++) {
+    const currentSegment = i < currentParts.length ? currentParts[i] : 0;
```

Comment:
that line and line below can have simplified condition: 
const currentSegment = currentParts[i] ?? 0;
This will be working because when you refer to element in array which doesn't exist then it will give just undefined so undefined ?? 0 will give 0